crmsh/utils.py

```diff
@@ -2292,4 +2149,255 @@ def interface_choice():
+        """
+        return IP(addr).ip_address in self.ip_interface.network
+
+
+class InterfacesInfo(object):
+    """
+    Class to collect interfaces information on local node
+    """
+
+    def __init__(self, ipv6=False, second_heartbeat=False, custom_nic_list=[]):
+        """
+        Init function
+
+        On init process,
+        "ipv6" is provided by -I option
+        "second_heartbeat" is provided by -M option
+        "custom_nic_list" is provided by -i option
+        """
+        self.ip_version = 6 if ipv6 else 4
+        self.second_heartbeat = second_heartbeat
+        self._default_nic_list = custom_nic_list
+        self._nic_info_dict = {}
+
+    def get_interfaces_info(self):
+        """
+        Try to get interfaces info dictionary via "ip" command
+        """
+        cmd = "ip -{} -o addr show".format(self.ip_version)
+        rc, out, err = get_stdout_stderr(cmd)
+        if rc != 0:
+            raise ValueError(err)
+
+        # format on each line will like:
+        # 2: enp1s0    inet 192.168.122.241/24 brd 192.168.122.255 scope global enp1s0\       valid_lft forever preferred_lft forever
+        for line in out.splitlines():
+            _, nic, _, ip_with_mask, *_ = line.split()
+            #TODO change this condition when corosync support link-local address
+            interface_inst = Interface(ip_with_mask)
+            if interface_inst.is_loopback or interface_inst.is_link_local:
+                continue
+            # one nic might configured multi IP addresses
+            if nic not in self._nic_info_dict:
+                self._nic_info_dict[nic] = []
+            self._nic_info_dict[nic].append(interface_inst)
+
+        if not self._nic_info_dict:
+            raise ValueError("No address configured")
+        if self.second_heartbeat and len(self._nic_info_dict) == 1:
+            raise ValueError("Cannot configure second heartbeat, since only one address is available")
+
+    @property
+    def nic_list(self):
+        """
+        Get interfaces name list
+        """
+        return list(self._nic_info_dict.keys())
+
+    @property
+    def interface_list(self):
+        """
+        Get instance list of class Interface
+        """
+        _interface_list = []
+        for interface in self._nic_info_dict.values():
+            _interface_list.extend(interface)
+        return _interface_list
+
+    @property
+    def ip_list(self):
+        """
+        Get IP address list
+        """
+        return [interface.ip for interface in self.interface_list]
+
+    @classmethod
+    def get_local_ip_list(cls, is_ipv6):
+        """
+        Get IP address list
+        """
+        cls_inst = cls(is_ipv6)
+        cls_inst.get_interfaces_info()
+        return cls_inst.ip_list
+
+    @property
+    def network_list(self):
+        """
+        Get network list
+        """
+        return list(set([interface.network for interface in self.interface_list]))
```

Comment:
Why do you do `list` and `set` operations? Just to understand what they are for

Reply:
To deduplicate the possible same network value